### PR TITLE
Implemented App Intro

### LINKED ISSUE #46: App Intro
**Describe the feature you'd like**
Currently, the app is using MaterialIntroView to tell the users about the features of the app. 
Now a view pager has to be shown in which various functionality of the app has to be shown to the user.
You can use this library to implement the same: https://github.com/AppIntro/AppIntro

**Additional context**
All the features highlighted using material intro view have to be shown in the view page based intro as well.
Remove the material intro view after adding this feature.
Values to be assigned when implementing the library(Have a look at documentation, you will get to know about these attributes):
Slide Transformer = AppIntroPageTransformerType.SlideOver
Wizard mode = true
Enabe ImmersiveMode
Also ask for "Do not distrub" permission in the intro itself. See the documentation, resources for permission are present there.

**Would you like to work on the issue?**

- [ ] Yes
- [ ] No
- Other: Open for all

### PR DESCRIPTION
**Pull request by Haid-Faiz**
Fixes #46 

Changes: Implemented App intro using App intro library... and old material intro features are now showing in app intro.

Screenshots for the change:

<table>
<tr>

<td>

</td>

<td>

</td>

<td>

</td>

</tr>
</table>

### CODE REVIEW

**Comment by Haid-Faiz** [comment]
@rob729  My gradle ktlint build is getting failed again and again.

Screenshot (188)

**Comment by Haid-Faiz** [comment]
@rob729 have a look... ktlint check fixed

File: app/src/main/java/com/rob729/quiethours/Activity/IntroScreen.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import com.github.appintro.AppIntro
+import com.github.appintro.AppIntroCustomLayoutFragment
+import com.github.appintro.AppIntroPageTransformerType
+import com.rob729.quiethours.R
+
+class IntroScreen : AppIntro() {
+
+    lateinit var sharedPreferences: SharedPreferences
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Activity/IntroScreen.kt]: StoreSession is being used for storing values in shared preferences.
Used it for the same.

File: app/src/main/java/com/rob729/quiethours/Activity/SplashScreen.kt [hunk 2]
```diff
@@ -24,4 +25,13 @@ class SplashScreen : AppCompatActivity() {
         }
         StoreSession.writeBoolean(AppConstants.FIRST_BOOT, true)
     }
+
+    fun checkAppIntro() {
+        val sharedPreferences = getSharedPreferences("AppIntroCheck", MODE_PRIVATE)
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Activity/SplashScreen.kt]: same comment as above.

File: app/src/main/java/com/rob729/quiethours/Activity/SplashScreen.kt [hunk 3]
```diff
@@ -16,6 +16,7 @@ class SplashScreen : AppCompatActivity() {
         startActivity(intent)
         overridePendingTransition(R.anim.fade_in, R.anim.fade_out)
         checkFirstRun()
+        checkAppIntro()
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Activity/SplashScreen.kt:16]: There is already a startActivity() present above checkFirstRun() and then you are adding similar code in checkAppIntro().
Make the changes for the proper working of the app.

**Comment by rob729** [comment]
@Haid-Faiz UI added is not optimized for dark mode as the text is not visible in dark mode.

**Comment by Haid-Faiz** [comment]
@rob729 ok i'll fix it...

**Comment by Haid-Faiz** [comment]
@rob729  Should i show status bar in app intro...?

**Comment by rob729** [comment]
> @rob729 Should i show status bar in app intro...?

No

**Comment by rob729** [comment]
@Haid-Faiz Also I didn't ask you to add a screenshot of the previous app intro in this one. You had to add the info about the earlier highlighted features.
The current images are not looking good. Maybe you can look into the apps which have implemented the same library to get a better idea. The list of apps using the same library is mentioned in the readme of the library.
Keep in mind to highlight the features which are offered in the current version like swipe to delete, click on the profile to get more details, pause/resume profile. Remove the add profile page, it seems to be too basic to show in app intro.

Just to give a little better idea, in case of the "click on the profile to get more details" feature you can add the screenshot of the details fragment bottom sheet. For the "swipe to delete" feature you can show a sample profile which is swiped towards the right little bit showing the delete icon with the red background.

File: app/build.gradle [hunk 4]
```diff
@@ -64,6 +64,9 @@ dependencies {
     def lottieVersion = '3.4.4'
     implementation "com.airbnb.android:lottie:$lottieVersion"
 
+    // App Intro Library
```
Comment (rob729) [inline app/build.gradle:67]: @Haid-Faiz Also remove the library of the previous app intro.

File: app/src/main/java/com/rob729/quiethours/Activity/IntroScreen.kt [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import com.rob729.quiethours.R
+import com.rob729.quiethours.util.AppConstants
+import com.rob729.quiethours.util.StoreSession
+
+class IntroScreen : AppIntro() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+//        showStatusBar(true)
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Activity/IntroScreen.kt]: remove the comment

File: app/src/main/java/com/rob729/quiethours/Activity/SplashScreen.kt [hunk 6]
```diff
@@ -12,8 +12,7 @@ class SplashScreen : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        val intent = Intent(this, MainActivity::class.java)
-        startActivity(intent)
+        checkAppIntro()
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Activity/SplashScreen.kt:16]: move the checkFirstRun() above checkAppIntro()

File: app/src/main/java/com/rob729/quiethours/util/StoreSession.kt [hunk 7]
```diff
@@ -61,4 +61,13 @@ object StoreSession {
     fun readLong(key: String): Long {
         return sharedPreferences.getLong(key, 0)
     }
+
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/util/StoreSession.kt]: There is already a function for writeBoolean and readBoolean then why add the below functions?

File: app/src/main/java/com/rob729/quiethours/Activity/SplashScreen.kt (hunk 6 above)
Comment (Haid-Faiz) [reply app/src/main/java/com/rob729/quiethours/Activity/SplashScreen.kt:16]: Ok

File: app/src/main/java/com/rob729/quiethours/util/StoreSession.kt (hunk 7 above)
Comment (Haid-Faiz) [reply app/src/main/java/com/rob729/quiethours/util/StoreSession.kt]: Ok.. i'll fix it

**Comment by rob729** [comment]
@Haid-Faiz In the second intro screen, crop out the app bar from screenshot

**Comment by Haid-Faiz** [comment]
@rob729 ok

**Comment by rob729** [comment]
@Haid-Faiz In the second app intro screen, it should be "Swipe left" instead of slide.
Use the screenshot from the latest version of the app as in the second screenshot the pause/resume toggle is missing.
Also, add the pause/resume profile feature in app intro.
Use roboto_slab font in the app intro.

**Comment by Haid-Faiz** [comment]
@rob729 ok

**Comment by rob729** [comment]
@Haid-Faiz The checks are failing. Fix them.

**Comment by Haid-Faiz** [comment]
@rob729  Ya.... i will fix.. i'll msg you when everything will completed

**Comment by Haid-Faiz** [comment]
@rob729 have a look now... & what is this codeClimate check....?? it is showing some similar code warning... How to pass this check ?

**Comment by rob729** [comment]
Click on the details present on the codeclimate check and it will show what are the problems present in the PR.

**Comment by Haid-Faiz** [comment]
@rob729  i have fixed 2 issues out of 4 of codeclimate but remaining 2 issues are not issues i think... because it is showing 2 similar block of code have found but actually one block is overridden method onSkipPressed() & second block is onDonePressed() present in IntroScreen Activity. So, now what should i do to fix this

**Comment by rob729** [comment]
> @rob729 i have fixed 2 issues out of 4 of codeclimate but remaining 2 issues are not issues i think... because it is showing 2 similar block of code have found but actually one block is overridden method onSkipPressed() & second block is onDonePressed() present in IntroScreen Activity. So, now what should i do to fix this

Since there is same code in both functions, move that block of code into a function. You can name the function to be navigateToMainFragment().

**Comment by rob729** [comment]
@Haid-Faiz One more thing, in the second app intro screen make it "swipe left a profile to delete it".
Also, make 'Y' capital of you.

**Comment by Haid-Faiz** [comment]
> > @rob729 i have fixed 2 issues out of 4 of codeclimate but remaining 2 issues are not issues i think... because it is showing 2 similar block of code have found but actually one block is overridden method onSkipPressed() & second block is onDonePressed() present in IntroScreen Activity. So, now what should i do to fix this
> 
> Since there is same code in both functions, move that block of code into a function. You can name the function to be navigateToMainFragment().

@rob729 Ya, fixed this but read your msg after fixing it.. i mean i named it appIntroClickListeners()

**Comment by Haid-Faiz** [comment]
> @Haid-Faiz One more thing, in the second app intro screen make it "swipe left a profile to delete it".
> Also, make 'Y' capital of you.

ok

File: app/src/main/java/com/rob729/quiethours/util/AppConstants.kt [hunk 8]
```diff
@@ -9,4 +9,5 @@ object AppConstants {
     val FIRST_BOOT = "FIRST_BOOT"
     val NIGHT_MODE = "nightMode"
     val RINGTONE_MODE = "RINGTONE_MODE"
+    val APP_INTRO_CHECK = "isIntroduced"
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/util/AppConstants.kt]: Make it val APP_INTRO_CHECK = "APP_INTRO_CHECK"

File: app/src/main/java/com/rob729/quiethours/Activity/IntroScreen.kt [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        super.onSkipPressed(currentFragment)
+        appIntroClickListeners()
+    }
+
+    override fun onDonePressed(currentFragment: Fragment?) {
+        super.onDonePressed(currentFragment)
+        appIntroClickListeners()
+    }
+
+    private fun appIntroClickListeners() {
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Activity/IntroScreen.kt]: Change function name to navigateToMainFragment()

**Review by rob729 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/rob729/quiethours/util/AppConstants.kt (hunk 8 above)
Comment (Haid-Faiz) [reply app/src/main/java/com/rob729/quiethours/util/AppConstants.kt]: ok

File: app/src/main/java/com/rob729/quiethours/Activity/IntroScreen.kt (hunk 9 above)
Comment (Haid-Faiz) [reply app/src/main/java/com/rob729/quiethours/Activity/IntroScreen.kt]: ok

Comment (Haid-Faiz) [reply app/src/main/java/com/rob729/quiethours/Activity/IntroScreen.kt]: @rob729 navigateToMainFragment() or navigateToMainActivity()    ...??

**Comment by Haid-Faiz** [comment]
@rob729  navigateToMainFragment()   OR navigateToMainActivity()

File: app/src/main/java/com/rob729/quiethours/Activity/IntroScreen.kt (hunk 9 above)
Comment (rob729) [reply app/src/main/java/com/rob729/quiethours/Activity/IntroScreen.kt]: Sorry, it should be Activity

Comment (Haid-Faiz) [reply app/src/main/java/com/rob729/quiethours/Activity/IntroScreen.kt]: ok

**Comment by Haid-Faiz** [comment]
@rob729 Now, have a look

**Review by rob729 — APPROVED** [review]

**Comment by rob729** [comment]
give @Haid-Faiz 75 points

**Pull request merged**